packages/amplify-provider-awscloudformation/src/utils/remove-dependent-function.ts

```diff
@@ -0,0 +1,68 @@
+import { $TSAny, $TSContext, $TSObject, pathManager } from 'amplify-cli-core';
+import path from 'path';
+import * as TransformPackage from 'graphql-transformer-core';
+
+export async function removeDependencyOnFunctions(
+  context: $TSContext,
+  apiResource: $TSObject,
+  allResources: $TSObject[],
+): Promise<$TSObject[]> {
+  // get #current-cloud-backed and cloud backend schema.graphql
+  let dependentFunctionResource;
+  const backendDir = pathManager.getBackendDirPath();
+  const currentBackendDir = pathManager.getCurrentCloudBackendDirPath();
+  const modelsDeleted = await getSchemaDiff(currentBackendDir, backendDir, apiResource[0].resourceName);
+  if (modelsDeleted.length) {
+    dependentFunctionResource = await context.amplify.invokePluginMethod(context, 'function', undefined, 'lambdasWithApiDependency', [
+      context,
+      allResources,
+      backendDir,
+      modelsDeleted,
+    ]);
+    if (dependentFunctionResource.length) {
+      const dependentFunctionsNames = dependentFunctionResource.map(lambda => lambda.resourceName);
+      context.print.info('');
+
+      context.print.warning(`Functions ${dependentFunctionsNames} have access to removed GraphQL API model(s) ${modelsDeleted}`);
+
+      const continueToPush = !!context?.exeInfo?.inputParams?.yes;
+
+      if (
+        !continueToPush &&
+        (await context.amplify.confirmPrompt('Do you want to remove the GraphQL model access on these affected functions?', false))
+      ) {
```

Comment:
I don't think this logic lines up with what I thought the expected DX is. What I thought was:
1. `amplify push` => gives customer the option to automatically remove function permissions. if they say no, then fail with manual update steps
2. `amplify push --yes` => fail with manual update steps
3. `amplify push --yes --force` => automatically remove function permissions and push

cc @renebrandel is this experience correct?